pkg/url/url_test.go

```diff
@@ -247,3 +266,121 @@ func TestExists(t *testing.T) {
 		}
 	}
 }
+
+func TestGetComponentServicePortNumbers(t *testing.T) {
+	type args struct {
+		componentName   string
+		applicationName string
+	}
+	tests := []struct {
+		name             string
+		args             args
+		selectors        string
+		returnedServices corev1.ServiceList
+		wantedPorts      []int
+		wantErr          bool
+	}{
+		{
+			name: "case 1: with valid values and one port",
+			args: args{
+				componentName:   "nodejs",
+				applicationName: "app",
+			},
+			selectors: "app.kubernetes.io/component-name=nodejs,app.kubernetes.io/name=app",
+			returnedServices: corev1.ServiceList{
+				Items: []corev1.Service{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app.kubernetes.io/name":           "app",
+								"app.kubernetes.io/component-name": "nodejs",
+							},
+						},
+						Spec: corev1.ServiceSpec{
+							Ports: []corev1.ServicePort{
+								{
+									Port: 8080,
+								},
+							},
+						},
+					},
+				},
+			},
+			wantedPorts: []int{8080},
+			wantErr:     false,
+		},
+		{
+			name: "case 2: with valid values and two ports",
+			args: args{
+				componentName:   "nodejs",
+				applicationName: "app",
+			},
+			selectors: "app.kubernetes.io/component-name=nodejs,app.kubernetes.io/name=app",
+			returnedServices: corev1.ServiceList{
+				Items: []corev1.Service{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app.kubernetes.io/name":           "app",
+								"app.kubernetes.io/component-name": "nodejs",
+							},
+						},
+						Spec: corev1.ServiceSpec{
+							Ports: []corev1.ServicePort{
+								{
+									Port: 8080,
+								},
+							},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app.kubernetes.io/name":           "app",
+								"app.kubernetes.io/component-name": "nodejs",
+							},
+						},
+						Spec: corev1.ServiceSpec{
+							Ports: []corev1.ServicePort{
+								{
+									Port: 9100,
+								},
+							},
+						},
+					},
+				},
+			},
+			wantedPorts: []int{8080, 9100},
+			wantErr:     false,
+		},
```

Comment:
Please also add error Case

Reply:
I don't think that a error case will be possible in this case as the error is thrown by GetServicesFromSelector() which calls List() and we are mocking it through the fake prependReactor

Reply:
Hmm, so there aren't any validations in GetComponentServicePortNumbers.
Ok so sounds good